src/test/unit/lib/model/OutboundTransfersModel.test.js

```diff
@@ -209,6 +209,99 @@ describe('outboundModel', () => {
+    });
+
+
+    test('uses quote response transfer amount for transfer prepare', async () => {
+        defaultEnv.AUTO_ACCEPT_PARTY = 'true';
+        defaultEnv.AUTO_ACCEPT_QUOTES = 'true';
+
+        await setConfig(defaultEnv);
+        const conf = getConfig();
+        const cache = new Cache();
+
+        MojaloopRequests.__getParties = jest.fn(() => {
+            cache.emitMessage(JSON.stringify(payeeParty));
+            return Promise.resolve();
+        });
+
+        // change the the transfer amount and currency in the quote response
+        // so it is different to the initial request
+        quoteResponse.data.transferAmount = {
+            currency: 'XYZ',
+            amount: '9876543210'
+        };
+
+        expect(quoteResponse.data.transferAmount).not.toEqual({
+            amount: transferRequest.amount,
+            currency: transferRequest.currency
+        });
+
+        MojaloopRequests.__postQuotes = jest.fn(() => {
+            // ensure that the `MojaloopRequests.postQuotes` method has been called with correct arguments
+            // including extension list
+            const extensionList = MojaloopRequests.__postQuotes.mock.calls[0][0].extensionList;
+            expect(extensionList).toBeTruthy();
+            expect(extensionList.extension).toBeTruthy();
+            expect(extensionList.extension.length).toBe(2);
+            expect(extensionList.extension[0]).toEqual({ key: 'qkey1', value: 'qvalue1' });
+            expect(extensionList.extension[1]).toEqual({ key: 'qkey2', value: 'qvalue2' });
+
+            // simulate a callback with the quote response
+            cache.emitMessage(JSON.stringify(quoteResponse));
+            return Promise.resolve();
+        });
+
+        MojaloopRequests.__postTransfers = jest.fn(() => {
+            //ensure that the `MojaloopRequests.postTransfers` method has been called with the correct arguments
+            // set as the destination FSPID, picked up from the header's value `fspiop-source`
+            expect(model.data.quoteResponseSource).toBe(quoteResponse.headers['fspiop-source']);
+
+            const postTransfersBody = MojaloopRequests.__postTransfers.mock.calls[0][0];
+
+            const extensionList = postTransfersBody.extensionList;
+            expect(extensionList.extension).toBeTruthy();
+            expect(extensionList.extension.length).toBe(2);
+            expect(extensionList.extension[0]).toEqual({ key: 'tkey1', value: 'tvalue1' });
+            expect(extensionList.extension[1]).toEqual({ key: 'tkey2', value: 'tvalue2' });
+
+            expect(MojaloopRequests.__postTransfers.mock.calls[0][1]).toBe(quoteResponse.headers['fspiop-source']);
+            expect(model.data.to.fspId).toBe(payeeParty.party.partyIdInfo.fspId);
+            expect(quoteResponse.headers['fspiop-source']).not.toBe(model.data.to.fspId);
+
+            expect(postTransfersBody.amount).toEqual(quoteResponse.data.transferAmount);
```

Comment:
I've just realized that this code block should also be placed after `await model.run();`, specifically after `.toHaveBeenCalledTimes` call.

Reply:
I dont think that can be done exactly @kirgene , these expects are checking the value passed to postTransfers.

Reply:
Yeah, these are the checks for `postTransfers` args, so I think it would be better to write something like this:
```
       MojaloopRequests.__postTransfers = jest.fn((body, destFspId) => {

...
            const extensionList = body.extensionList;
...
            expect(destFspId).toBe(quoteResponse.headers['fspiop-source']);
...
   });
 ```